packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -442,6 +444,74 @@ describe('getPathFromSpecPattern', () => {
   })
 })
 
+describe('_makeSpecWatcher', () => {
+  let ctx: DataContext
+  let specWatcher: chokidar.FSWatcher
+
+  beforeEach(function () {
+    this.timeout(20000) // FixturesHelper.remove can take awhile depending on what you've done prior
+    FixturesHelper.remove()
+
+    ctx = createTestDataContext('open')
+
+    FixturesHelper.scaffold('spec-watcher')
+
+    this.specWatcherPath = FixturesHelper.projectPath('spec-watcher')
+
+    ctx.actions.project.setCurrentProjectAndTestingTypeForTestSetup(this.specWatcherPath)
+  })
+
+  afterEach(async () => {
+    sinon.restore()
+    await specWatcher.close()
+    ctx.destroy()
+  })
+
+  it('watch for changes on files based on the specPattern', async function () {
+    function delay (ms) {
+      return new Promise((resolve) => {
+        setTimeout(resolve, ms)
+      })
+    }
+
+    specWatcher = ctx.project._makeSpecWatcher({
+      projectRoot: this.specWatcherPath,
+      specPattern: ['**/*.{cy,spec}.{ts,js}'],
+      excludeSpecPattern: ['**/ignore.spec.ts'],
+      additionalIgnorePattern: ['additional.ignore.cy.js'],
+    })
+
+    await new Promise((resolve) => {
+      specWatcher.once('ready', resolve)
+    })
+
+    const allFiles = new Set()
+
+    specWatcher.on('add', (filePath) => {
+      allFiles.add(filePath)
+    })
+
+    specWatcher.on('change', (filePath) => {
+      allFiles.add(filePath)
+    })
+
+    const SUPPORT_FILE = path.join(this.specWatcherPath, 'cypress', 'support', 'e2e.js')
+    const SPEC_FILE1 = path.join(this.specWatcherPath, 'cypress', 'e2e', 'foo.cy.js')
+    const SPEC_FILE2 = path.join(this.specWatcherPath, 'cypress', 'e2e', 'some', 'new', 'folder', 'foo.cy.js')
+
+    ctx.actions.file.writeFileInProject(SUPPORT_FILE, '// foo')
+    ctx.actions.file.writeFileInProject(SPEC_FILE1, '// foo')
+    ctx.actions.file.writeFileInProject(SPEC_FILE2, '// foo')
+
+    await delay(500)
+    expect(Array.from(allFiles)).to.eql([
+      SPEC_FILE1,
+      SPEC_FILE2,
+    // Paths don't start with '/' when emitted
+    ].map((p) => p.slice(1)))
+  })
```

Comment:
So is the assertion here that we "aren't watching `SUPPORT_FILE`"? I think a comment or an explicit assertion would make it clearer:

```suggestion
    // by the end, we should only be watching specs, not support
    expect(Array.from(allFiles)).to.eql([
      SPEC_FILE1,
      SPEC_FILE2,
    // Paths don't start with '/' when emitted
    ].map((p) => p.slice(1)))
  })
```

Reply:
Tests were made more descriptive in another way, resolved